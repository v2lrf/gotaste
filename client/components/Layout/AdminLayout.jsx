import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import Head from 'next/head'
import Link from 'next/link'

import Container from '../Container'
import NavItem from '../NavBar/NavItem'

function AdminLayout({ title, children }) {
  return (
    <Fragment>
      <Head>
        <title>{title}</title>
      </Head>
      <div className="bg-white">
        <Container>
          <div className="h-auto sm:h-16 flex items-center justify-between">
            <div className="my-4 sm:my-0">
              <Link href="/admin">
                <a className="flex text-red-light text-xl font-semibold no-underline">
                  <img
                    src="/static/gotaste-120.png"
                    alt="GoTaste"
                    className="h-6 pr-2"
                  />
                  GoTaste Admin
                </a>
              </Link>
            </div>
            <ul className="list-reset flex flex-row">
              <NavItem asActive to="/admin">
                Overblik
              </NavItem>
              <NavItem asActive to="/admin/businesses">
                Forretninger
              </NavItem>
              <div className="mx-2 my-1 border border-red-lightest" />
              <NavItem to="/">GoTaste</NavItem>
              <NavItem to="/sign-out">Log ud</NavItem>
            </ul>
          </div>
        </Container>
      </div>
      {children}
    </Fragment>
  )
}

AdminLayout.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node.isRequired
}

AdminLayout.defaultProps = {
  title: 'GoTaste Admin'
}

export default AdminLayout
