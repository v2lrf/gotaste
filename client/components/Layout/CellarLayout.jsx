import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import Head from 'next/head'
import Link from 'next/link'

import Container from '../Container'
import NavItem from '../NavBar/NavItem'

function CellarLayout({ title, children }) {
  return (
    <Fragment>
      <Head>
        <title>{title}</title>
      </Head>
      <div className="bg-white">
        <Container>
          <div className="h-auto sm:h-16 flex items-center justify-between">
            <div className="my-4 sm:my-0">
              <Link href="/">
                <a className="flex text-red-light text-xl font-semibold no-underline">
                  <img
                    src="/static/gotaste-120.png"
                    alt="GoTaste"
                    className="h-6 pr-2"
                  />
                  GoTaste Cellar
                </a>
              </Link>
            </div>
            <ul className="list-reset flex flex-row">
              <NavItem asActive to="/cellar">
                Overblik
              </NavItem>
              <NavItem asActive to="#">
                Analytics
              </NavItem>
              <NavItem asActive to="#">
                Begivenheder
              </NavItem>
              <NavItem asActive to="#">
                Anmeldelser
              </NavItem>
              <NavItem asActive to="/cellar/profile">
                Profil
              </NavItem>
              <div className="mx-2 my-1 border border-red-lightest" />
              <NavItem to="/sign-out">Log ud</NavItem>
            </ul>
          </div>
        </Container>
      </div>
      {children}
    </Fragment>
  )
}

CellarLayout.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node.isRequired
}

CellarLayout.defaultProps = {
  title: 'GoTaste Cellar'
}

export default CellarLayout
