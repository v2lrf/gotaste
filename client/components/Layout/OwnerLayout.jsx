import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import Head from 'next/head'
import Link from 'next/link'

import Container from '../Container'
import NavItem from '../NavBar/NavItem'

function OwnerLayout({ title, children }) {
  return (
    <Fragment>
      <Head>
        <title>{title}</title>
      </Head>
      <div className="bg-white shadow-lg">
        <Container>
          <div className="h-auto sm:h-16 flex items-center justify-between">
            <div className="my-4 sm:my-0">
              <Link href="/">
                <a className="flex text-red-light text-xl font-semibold no-underline">
                  <img
                    src="/static/gotaste-120.png"
                    alt="GoTaste"
                    className="h-6 pr-2"
                  />
                  GoTaste Cellar
                </a>
              </Link>
            </div>
            <ul className="list-reset flex flex-row">
              <NavItem to="#">Overblik</NavItem>
              <NavItem to="#">Analytics</NavItem>
              <NavItem to="#">Begivenheder</NavItem>
              <NavItem to="#">Anmeldelser</NavItem>
              <NavItem to="#">Profil</NavItem>
              <div className="mx-2 my-1 border border-red-lightest" />
              <NavItem to="/sign-out">Log ud</NavItem>
            </ul>
          </div>
        </Container>
      </div>
      {children}
    </Fragment>
  )
}

OwnerLayout.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node.isRequired
}

OwnerLayout.defaultProps = {
  title: 'GoTaste Cellar'
}

export default OwnerLayout
