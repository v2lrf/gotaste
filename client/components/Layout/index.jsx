import React, { Fragment } from 'react'
import PropTypes from 'prop-types'
import Head from 'next/head'

import config from '../../config'

import NavBar from '../NavBar'
import Footer from '../Footer'

function Layout({ children, title, metaDescription, metaKeywords }) {
  const keywords = config.DEFAULT_META_KEYWORDS.concat(metaKeywords).join(',')
  return (
    <Fragment>
      <Head>
        <title>{title}</title>
        <meta name="description" content={metaDescription} />
        <meta name="keywords" content={keywords} />
      </Head>
      <NavBar />
      {children}
      <Footer />
    </Fragment>
  )
}

Layout.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string,
  metaDescription: PropTypes.string,
  metaKeywords: PropTypes.arrayOf(PropTypes.string)
}

Layout.defaultProps = {
  title: 'Govinu - Smag og gå på opdagelse i vinens verden',
  metaDescription:
    'Find og følg din lokale vinforhandler eller vinbar. Gå på opdagelse i vinens verden, find nye smagninger eller arrangementer og meget mere.',
  metaKeywords: ['']
}

export default Layout
