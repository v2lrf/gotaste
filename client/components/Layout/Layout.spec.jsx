import React from 'react'
import Layout from '.'

describe('Footer', () => {
  let component
  beforeEach(() => {
    component = shallow(<Layout>Some content!</Layout>)
  })

  describe('Snapshots', () => {
    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
