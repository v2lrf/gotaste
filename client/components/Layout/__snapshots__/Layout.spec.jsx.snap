// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Snapshots should match the snapshot 1`] = `
<Fragment>
  <Head>
    <title>
      Govinu - Smag, og gå på opdagelse i vinens verden
    </title>
    <meta
      content="Find, og følg din lokale vinforhandler eller vinbar. Gå på opdagelse i vinens verden, find nye smagninger eller arrangementer, og meget mere."
      name="description"
    />
    <meta
      content="govinu,vin,god vin,vinsmagning,blindsmagning,"
      name="keywords"
    />
  </Head>
  <NavBar />
  Some content!
  <Footer />
</Fragment>
`;
